### PR TITLE
UCT/ADDRESS: Declare address as packed, to fix valgrind error.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1665/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1053/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1055/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1667/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1057/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1669/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1058/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1671/ for details.

**Comment by yosefe** [comment]
@shamisp pls review

**Comment by shamisp** [comment]
:+1:

File: src/ucp/core/ucp_worker.c [hunk 1]
```diff
@@ -488,6 +496,8 @@ ucs_status_t ucp_worker_get_address(ucp_worker_h worker, ucp_address_t **address
             goto err_free;
         }
 
+        VALGRIND_CHECK_MEM_IS_DEFINED(rsc_addr, rsc_addr_length);
+
```
Comment (artpol84) [inline src/ucp/core/ucp_worker.c]: @yosefe,
I'm not familiar with Valgrind enough, so can you explain how the line above works? As I was able to understand `VALGRIND_CHECK_MEM_IS_DEFINED` returns the pointer on the first undefined byte. But you don't check for it.
Maybe you meant `VALGRIND_MAKE_MEM_DEFINED` ?

Comment (yosefe) [inline src/ucp/core/ucp_worker.c]: @artpol84  `VALGRIND_CHECK_MEM_IS_DEFINED` does not only return the first undefined byte, it also prints an error (same way OpenMPI does it, with its MEMCHECKER calls). So here I want to generate an error as early as possible if the address is undefined, and not wait until, for example, OpenMPI's modex code checks it.

**Pull request merged**